Modules/Quotas/test/Quotas.Application.Tests/TestDoubles/FindAllWithKeysMetricsStubRepository.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Backbone.Modules.Quotas.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Quotas.Domain.Aggregates.Metrics;
+
+namespace Backbone.Modules.Quotas.Application.Tests.TestDoubles;
+public class FindAllWithKeysMetricsStubRepository : IMetricsRepository
+{
+    private readonly List<Metric> _metrics;
+
+    public FindAllWithKeysMetricsStubRepository(List<Metric> metrics)
+    {
+        _metrics = metrics;
+    }
+
+    public Task<Metric> Find(MetricKey key, CancellationToken cancellationToken, bool track = false)
+    {
+        throw new NotImplementedException();
+    }
+
+    public Task<IEnumerable<Metric>> FindAll(CancellationToken cancellationToken, bool track = false)
+    {
+        throw new NotImplementedException();
+    }
+
+    public Task<IEnumerable<Metric>> FindAllWithKeys(IEnumerable<MetricKey> keys, CancellationToken cancellationToken, bool track = false)
+    {
+        return Task.FromResult(_metrics.AsEnumerable());
```

Comment:
```suggestion
        return Task.FromResult(_metrics);
```

Reply:
You could by changing the types of the constructor parameter and the field to IEnumerable